### PR TITLE
cleanup and enhance validate_accounting.py

### PR DESCRIPTION
**Pull request by brianolson**
## Summary

drop algod sqlite3 support
add indexer api token support

See #312 

## Test Plan

Run this in `e2elive.py` short tests and also run it against a full mainnet indexer&algod.

### CODE REVIEW

**Review by egieseke — COMMENTED** [review]
Has this been tested with a remote node?  I have not been able to get it to work. 

Traceback (most recent call last):
  File "./scripts/../indexer/misc/validate_accounting.py", line 762, in <module>
    sys.exit(main())
  File "./scripts/../indexer/misc/validate_accounting.py", line 699, in main
    i2a_checker = check_from_algod(args)
  File "./scripts/../indexer/misc/validate_accounting.py", line 613, in check_from_algod
    process_genesis(algod.algod_request("GET", "/genesis"))

**Comment by egieseke** [comment]
Using a local genesis file, I was able to run the validate_accounting.py script.

**Comment by brianolson** [comment]
Remote access requires
https://github.com/algorand/py-algorand-sdk/pull/177 for algod /genesis api support in py-algorand-sdk

**Review by egieseke — APPROVED** [review]
LGTM

File: misc/validate_accounting.py [hunk 1]
```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
...
-                    emsg = '{} algod has assets but not indexer: {!r}\n'.format(niceaddr, nonzero)
-                    xe(emsg)
-            i2acfg = i2v.get('created-assets')
-            # filter out deleted entries that indexer is showing to us
+                    xe('{} indexer has assets but not algod: {!r}\n'.format(niceaddr, nonzero))
+        elif assets:
+            allzero = True
+            nonzero = []
+            for assetrec in assets:
+                if assetrec.get('amount',0) != 0:
```
Comment (winder) [inline misc/validate_accounting.py:412]: you can use the 'deleted' flag for this now, slightly more correct since an asset with 0 balance is still valid.

File: misc/validate_accounting.py [hunk 2]
```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
...
+        address = algosdk.encoding.decode_address(niceaddr)
+        xe = ccerr(self.err)
 
-    def check(self, address, niceaddr, microalgos, assets, acfg, appparams, applocal):
-        # check data from sql or algod vs indexer
-        i2v = self.accounts.pop(address, None)
-        if i2v is None:
-            self.neq += 1
-            self.err.write('{} not in indexer\n'.format(niceaddr))
+        microalgos = algod_account['amount-without-pending-rewards']
```
Comment (winder) [inline misc/validate_accounting.py:381]: missing a few top level fields:
- apps-local-state (not supported by indexer yet)
- rewards (needs some special handling but that is fixed in develop)
- status (need to remove spaces from the algod status)

File: misc/validate_accounting.py [hunk 3]
```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
...
-                    if not allzero:
-                        xe('{} indexer has assets but not algod: {!r}\n'.format(niceaddr, nonzero))
-            elif assets:
                 allzero = True
                 nonzero = []
-                for assetrec in assets:
-                    if assetrec.get('amount',0) != 0:
-                        allzero = False
+                for assetrec in i2assets:
+                    if assetrec.get('amount', 0) != 0:
```
Comment (winder) [inline misc/validate_accounting.py:403]: Use 'deleted'

File: misc/validate_accounting.py [hunk 4]
```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
...
-                pass # still ok
+            emsg = 'algod v={} i2 v={} '.format(microalgos, indexerAlgos)
+            if address == reward_addr:
+                emsg += ' Rewards account'
+            elif address == fee_addr:
+                emsg += ' Fee account'
+            xe(emsg)
+        assets = algod_account.get('assets')
+        i2assets = indexer_account.get('assets')
+        if i2assets:
```
Comment (winder) [inline misc/validate_accounting.py:394]: the `if{ if/else } else` logic is hard to follow. Now that we have the deleted flag can you filter the deleted fields and just compare lengths?

File: misc/validate_accounting.py [hunk 5]
```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
...
+            if assets:
+                emsg = assetEquality(i2assets, assets)
```
Comment (winder) [inline misc/validate_accounting.py:396]: `assetEquality` doesn't seem to check `default-frozen`, maybe another dictify function could strip out the indexer-only fields and do the deep equal?

File: misc/validate_accounting.py [hunk 6]
```diff
@@ -599,139 +548,212 @@ def token_addr_from_args(args):
...
-    for address in i2a.keys():
-        niceaddr = algosdk.encoding.encode_address(address)
-        rawad.append(algod.account_info(niceaddr))
-    aloadtime = time.time() - aloadstart
-    logger.info('loaded %d accounts from algod in %.1f seconds, %.1f a/s', len(rawad), aloadtime, len(rawad)/aloadtime)
+    indexer_headers = None
+    gtaddr = args.gtaddr
+    maxaddr = None
+    if args.shard:
+        minaddr, maxaddr = shard_bounds(args.shard)
```
Comment (winder) [inline misc/validate_accounting.py:619]: I found that postgres could dump all 8+ million accounts in 8 seconds, so this stuff may not be necessary. Just grab all the accounts and divvy them up.

**Review by winder — COMMENTED** [review]
This program is pretty hard for me to follow. I hope this feedback is still useful.
